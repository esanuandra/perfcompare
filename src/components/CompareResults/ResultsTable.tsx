import Box from '@mui/material/Box';
import CircularProgress from '@mui/material/CircularProgress';
import Paper from '@mui/material/Paper';
import { style } from 'typestyle';

import { useAppSelector } from '../../hooks/app';
import { selectProcessedResults } from '../../reducers/ComparisonSlice';
import { Colors, Spacing } from '../../styles';
import { ThemeMode } from '../../types/state';
import NoResultsFound from './NoResultsFound';
import TableContent from './TableContent';
import TableHeader from './TableHeader';

const customStyles = {
  boxShadow: 'none',
};

function ResultsTable(props: ResultsTableProps) {
  const { themeMode } = props;

  const loading = useAppSelector((state) => state.compareResults.loading);

  const processedResults = useAppSelector(selectProcessedResults);

  const themeColor100 =
    themeMode === 'light' ? Colors.Background100 : Colors.Background100Dark;

  const styles = {
    tableContainer: style({
      backgroundColor: themeColor100,
      marginTop: Spacing.Large,
      paddingBottom: Spacing.Large,
    }),
  };

  return (
    <Paper
      className={styles.tableContainer}
      data-testid='results-table'
      sx={customStyles}
      role='table'
    >
      {loading ? (
        <Box display='flex' justifyContent='center' alignItems='center'>
          <CircularProgress />
        </Box>
      ) : (
        <>
          <TableHeader themeMode={themeMode} />
          {processedResults.map((res, index) => (
            <TableContent
              themeMode={themeMode}
              key={index}
              header={res.revisionHeader}
              results={res.value}
            />
          ))}
        </>
      )}
      {!loading && processedResults.length == 0 && <NoResultsFound />}
    </Paper>
  );
}

interface ResultsTableProps {
  themeMode: ThemeMode;
}

export default ResultsTable;
