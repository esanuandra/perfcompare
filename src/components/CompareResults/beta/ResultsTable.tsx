import Paper from '@mui/material/Paper';
import Table from '@mui/material/Table';
import TableContainer from '@mui/material/TableContainer';
import { style } from 'typestyle';

import { Spacing } from '../../../styles';
import TableContent from './TableContent';
import TableHeader from './TableHeader';

const styles = {
  tableContainer: style({
    marginTop: Spacing.Large,
  }),
};

function ResultsTable() {
  return (
    <TableContainer component={Paper} className={styles.tableContainer} data-testid='results-table'>
      <Table>
        <TableHeader />
        {/* TODO: Add table body */}
        <TableContent />
      </Table>
    </TableContainer>
  );
}

export default ResultsTable;
