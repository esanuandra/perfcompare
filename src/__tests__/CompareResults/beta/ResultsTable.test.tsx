import ResultsTable from '../../../components/CompareResults/beta/ResultsTable';
import { renderWithRouter } from '../../utils/setupTests';
import { screen } from '../../utils/test-utils';

describe('Results Table', () => {
  it('Should match snapshot', () => {
    renderWithRouter(<ResultsTable />);

    expect(screen.getByTestId('results-table')).toBeInTheDocument();
    expect(document.body).toMatchSnapshot();
  });
});