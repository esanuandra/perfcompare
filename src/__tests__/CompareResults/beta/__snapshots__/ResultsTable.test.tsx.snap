// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Table Should match snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root f1otqxop css-pfrgnn-MuiPaper-root-MuiTableContainer-root"
      data-testid="results-table"
    >
      <table
        aria-label="collapsible table"
        class="MuiTable-root css-yso468-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root f1wbvtga css-15wwp11-MuiTableHead-root"
          data-testid="table-header"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1yt2i5z-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium filter css-hwty3m-MuiTableCell-root"
              scope="col"
            >
              Platform
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-hwty3m-MuiTableCell-root"
              scope="col"
            >
              Base
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-hwty3m-MuiTableCell-root"
              scope="col"
            >
              New
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium filter css-hwty3m-MuiTableCell-root"
              scope="col"
            >
              Status
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-hwty3m-MuiTableCell-root"
              scope="col"
            >
              Delta(%)
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium filter css-hwty3m-MuiTableCell-root"
              scope="col"
            >
              Confidence
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-hwty3m-MuiTableCell-root"
              scope="col"
            >
              Total Runs
            </th>
          </tr>
        </thead>
      </table>
    </div>
  </div>
</body>
`;
